OpenTokMeet/app/src/main/java/com/opentok/android/meet/HomeActivity.java

```diff
@@ -61,84 +75,201 @@ public void onCreate(Bundle savedInstanceState) {
+                } else {
+                    /* report error */
+                    (new AlertDialog.Builder(HomeActivity.this))
+                            .setTitle(R.string.error_title)
+                            .setMessage(R.string.error)
+                            .setCancelable(false)
+                            .setPositiveButton(
+                                    "OK",
+                                    new DialogInterface.OnClickListener() {
+                                        public void onClick(DialogInterface dialog, int id) {
+                                            finish();
+                                        }
+                                    })
+                            .create().show();
+                }
+            }
+        };
+        /* update values from ui */
+        mRoomName   = ((EditText)findViewById(R.id.input_room_name)).getText().toString();
+        mUsername   = ((EditText)findViewById(R.id.input_username)).getText().toString();
+        mH264Support= ((Switch)findViewById(R.id.h264Support)).isChecked();
+        /* save conference settings */
+        saveSettings();
+        /* set debug settings (setup from advanced settings menu) */
+        updatePreferences();
+        /* request conference information from server */
+        fetchRoomData.execute(mRoomName);
     }
 
-    private void saveConferenceData() {
-
-        SharedPreferences settings = getApplicationContext()
-                .getSharedPreferences(LAST_CONFERENCE_DATA, 0);
-        SharedPreferences.Editor editor = settings.edit();
-        editor.putString("roomName", roomName);
-        editor.putString("username", username);
-
+    private void saveSettings() {
+        SharedPreferences.Editor editor     = null;
+        /* Conference Settings */
+        editor = this.getSharedPreferences(PREF_TAG_CONFERENCE, 0).edit();
+        editor.putString(PREF_ROOMNAME, mRoomName);
+        editor.putString(PREF_USERNAME, mUsername);
+        editor.putString(PREF_CAPTURERRES, mCapturerResolution);
+        editor.putString(PREF_CAPTURERFPS, mCapturerFps);
+        editor.putBoolean(PREF_H264, mH264Support);
         editor.apply();
     }
 
-    private void restoreConferenceData() {
-        SharedPreferences settings = getApplicationContext()
-                .getSharedPreferences(LAST_CONFERENCE_DATA, 0);
-        roomName = settings.getString("roomName", "");
-        username = settings.getString("username", "");
+    private void updatePreferences() {
+        SharedPreferences settings = this.getSharedPreferences(PREF_TAG_DEBUG, 0);
+        OpenTokConfig.setJNILogs(settings.getBoolean(PREF_JNI_LOG, false));
+        OpenTokConfig.setOTKitLogs(settings.getBoolean(PREF_OTK_LOG, false));
+        OpenTokConfig.setOTKitLogs(settings.getBoolean(PREF_RTC_LOG, false));
+        OpenTokConfig.setUseMediaCodecFactories(settings.getBoolean(PREF_MEDIACODEC, false));
```

Comment:
Okay, gotcha. We set the h264 setting here. No, h264 support setting implies using media factories.